### PR TITLE
Wallet create tests

### PR DESCRIPTION
**Pull request by piotr-iohk**
# Issue Number

# Overview

- [x] I have added tests for wallet creation focusing on wallet name and mnemonic_sentence
- [ ] Still to add more test touching on other parameters (mnemonic_second_factor, passphrase, poolgap) and some more negative tests producing various error messages...

# Comments

### CODE REVIEW

File: src/Cardano/Wallet/Primitive/Types.hs [hunk 1]
```diff
@@ -168,7 +168,7 @@ instance FromText WalletName where
         | T.length t < walletNameMinLength =
             Left $ TextDecodingError $
                 "name is too short: expected at least "
-                    <> show walletNameMinLength <> " chars"
+                    <> show walletNameMinLength <> " char"
```
Comment (KtorZ) [inline src/Cardano/Wallet/Primitive/Types.hs]: _characters_ ?

File: test/integration/Test/Integration/Scenario/Wallets.hs [hunk 2]
```diff
@@ -102,9 +109,246 @@ spec = do
         r2 <- request @ApiWallet ctx ("POST", "v2/wallets") Default payload
         expectResponseCode @IO HTTP.status409 r2
 
+    describe "WALLETS_CREATE_04 - Wallet name" $ do
+        let walNameMax = T.pack (replicate walletNameMaxLength 'ą')
+        let matrix = [ ( show walletNameMinLength ++ " char long"
```
Comment (KtorZ) [inline test/integration/Test/Integration/Scenario/Wallets.hs]: Careful with the variable-length indentation here (and below) :pray:

File: test/integration/Test/Integration/Scenario/Wallets.hs [hunk 3]
```diff
@@ -102,9 +109,246 @@ spec = do
...
+                } |]
+        r <- request @ApiWallet ctx ("POST", "v2/wallets") Default payload
+        verify r
+            [ expectResponseCode @IO HTTP.status400
+            , expectErrorMessage "key \"name\" not present"
+            ]
+
+    describe "WALLETS_CREATE_05 - Mnemonics" $ do
+            let matrix = [ ( "[] as mnemonic_sentence -> fail"
+                           , []
```
Comment (KtorZ) [inline test/integration/Test/Integration/Scenario/Wallets.hs]: Same here, indentation please :grimacing:

File: test/integration/Test/Integration/Scenario/Wallets.hs [hunk 4]
```diff
@@ -102,9 +109,246 @@ spec = do
...
+                           , [ expectResponseCode @IO HTTP.status400
+                             , expectErrorMessage "ErrMnemonicWords (ErrWrongNumberOfWords 0 24)"
+                             ]
+                           )
+                         , ( "specMnemonicSentence -> fail"
+                           , specMnemonicSentence
+                           , [ expectResponseCode @IO HTTP.status400
+                             , expectErrorMessage "ErrMnemonicWords (ErrWrongNumberOfWords 15 24)"
+                             ] -- probably need to modify after bug #192 fixed
```
Comment (KtorZ) [inline test/integration/Test/Integration/Scenario/Wallets.hs]: Indeed. Let's get #192 resolved today and merge this one after in its corrected form :+1:

File: test/integration/Test/Integration/Scenario/Wallets.hs [hunk 5]
```diff
@@ -102,9 +110,427 @@ spec = do
...
+               )
+             , ( "invalid mnemonics -> fail", invalidMnemonics15
+               , [ expectResponseCode @IO HTTP.status400
+                 , expectErrorMessage "Invalid entropy checksum: please \
+                      \double-check the last word of your mnemonic sentence."
+                 ]
+               )
+             , ( "Japanese mnemonics -> fail", japaneseMnemonics15
+               , [ expectResponseCode @IO HTTP.status400
+                 , expectErrorMessage "Found invalid (non-English) word: "
```
Comment (piotr-iohk) [inline test/integration/Test/Integration/Scenario/Wallets.hs]: @KtorZ here `Found invalid (non-English) word:` only for Japanese mnemonics, for Chinese and French throws `Invalid number of words`. Why is that?

File: test/integration/Test/Integration/Scenario/Wallets.hs [hunk 6]
```diff
@@ -102,9 +110,427 @@ spec = do
...
+                   , [ expectResponseCode @IO HTTP.status400
+                     , expectErrorMessage "Invalid entropy checksum: please\
+                         \ double-check the last word of your mnemonic sentence."
+                     ]
+                   )
+                 , ( "invalid mnemonics -> fail", invalidMnemonics12
+                   , [ expectResponseCode @IO HTTP.status400
+                     , expectErrorMessage "Invalid number of words: 9 or 12\
+                         \ words are expected."
+                     ] -- why not Invalid entropy checksum ?
```
Comment (piotr-iohk) [inline test/integration/Test/Integration/Scenario/Wallets.hs]: here I was expecting `Invalid entropy checksum`. The mnemonic list is invalid...

File: test/integration/Test/Integration/Scenario/Wallets.hs [hunk 7]
```diff
@@ -102,9 +110,427 @@ spec = do
...
+                 , ( "Japanese mnemonics -> fail", japaneseMnemonics12
+                   , [ expectResponseCode @IO HTTP.status400
+                     , expectErrorMessage "Invalid number of words: 9 or 12\
+                          \ words are expected."
+                     ]
+                   )
+                 , ( "Chinese mnemonics -> fail", chineseMnemonics9
+                   , [ expectResponseCode @IO HTTP.status400
+                     , expectErrorMessage "Found invalid (non-English) word:"
+                     ] -- why only for Chinese?
```
Comment (piotr-iohk) [inline test/integration/Test/Integration/Scenario/Wallets.hs]: again `Found invalid (non-English) word:`. This time for Chinese mnemonics...
for Japanese and French it throws `Invalid number of words:`

**Review by piotr-iohk — COMMENTED** [review]
@KtorZ I have aligned the test after #197. In some cases there are some different than expected errors thrown. Please have a look at the comments inline...

I have also added cases for `passphrase` and `mnemonic_second_factor`. I want to add more tests for address_pool_gap and some destructive scenarios, but maybe I can add it on another PR - not to make this one too big.

File: test/integration/Test/Integration/Scenario/Wallets.hs [hunk 8]
```diff
@@ -102,9 +112,504 @@ spec = do
...
+                  )
+                , ( "0 -> fail", 0
+                   , [ expectResponseCode @IO HTTP.status400
+                     , expectErrorMessage "ErrGapOutOfRange 0"
+                     ] -- probably to be modified after #204
+                  )
+                , ( "-1 -> fail", (-1)
+                  , [ expectResponseCode @IO HTTP.status400
+                    , expectErrorMessage "ErrGapOutOfRange 255"
+                    ] -- probably to be modified after #204
```
Comment (piotr-iohk) [inline test/integration/Test/Integration/Scenario/Wallets.hs]: @KtorZ - I pushed tests for `address_pool_gap` here after all. Also created #204 as discussed.

**Review by KtorZ — APPROVED** [review]
LGTM, merging as is, will add more tests in additional PRs (and fix error messages after #204)
